apps/mobile/src/components/InsideRouter/components/SettingsScreen/atoms.ts

```diff
@@ -2,15 +2,39 @@ import {
   userPhoneNumberAtom,
   userDataRealOrAnonymizedAtom,
 } from '../../../../state/session'
-import {atom} from 'jotai'
+import {atom, type SetStateAction, type WritableAtom} from 'jotai'
 import {type ImportContactFromLinkPayload} from '../../../../state/contacts/domain'
 import {screenshotsDisabledAtom} from '../../../../state/showYouDidNotAllowScreenshotsActionAtom'
+import {focusAtom} from 'jotai-optics'
+import {preferencesAtom} from '../../../../utils/preferences'
+import getValueFromSetStateActionOfAtom from '../../../../utils/atomUtils/getValueFromSetStateActionOfAtom'
 
 export const reportIssueDialogVisibleAtom = atom<boolean>(false)
 export const changeCurrencyDialogVisibleAtom = atom<boolean>(false)
 export const qrCodeDialogVisibleAtom = atom<boolean>(false)
 export const qrScannerDialogVisibleAtom = atom<boolean>(false)
 
+export const currentAppLanguageAtom = focusAtom(preferencesAtom, (o) =>
```

Comment:
Please move to utils/preferences file so it's it is defined in the same place as other similar atoms are